xcube/core/extract.py

```diff
@@ -314,38 +408,56 @@ def get_dataset_indexes(dataset: xr.Dataset,
     return interp_indexes
 
 
-def _empty_points_var(var: xr.DataArray, num_points):
+def _empty_points_var(var: xr.DataArray, num_points: int):
     fill_value = 0 if np.issubdtype(var.dtype, np.integer) else np.nan
     return xr.DataArray(np.full(num_points, fill_value, dtype=var.dtype),
                         dims=[INDEX_DIM_NAME], attrs=var.attrs)
 
 
-def _normalize_points(points: Union[xr.Dataset, pd.DataFrame, Mapping[str, Any]]) -> xr.Dataset:
-    if isinstance(points, pd.DataFrame):
-        points = xr.Dataset.from_dataframe(points)
-    elif not isinstance(points, xr.DataArray):
-        points = xr.Dataset(points)
-    return points
-
+def _normalize_series(
+        points: PointsLike,
+        col_names: Sequence[str],
+        force_dataset: bool = False,
+        param_name: str = "points"
+) -> Tuple[PointsLike, int]:
+    if not isinstance(points, (xr.Dataset, pd.DataFrame)):
+        new_points = {}
+        for col_name in col_names:
+            if col_name in points:
+                col = points[col_name]
+                if not isinstance(col, (np.ndarray,
```

Comment:
Did you forget an else here?

Reply:
No, this is a filter. Later below col_names will be validated. This is more or less old code.